cppapi/client/devapi_attr.cpp

```diff
@@ -1116,6 +1118,7 @@ DeviceAttribute::DeviceAttribute(string& new_name, vector<short> &datum):ext(new
 	exceptions_flags.set(isempty_flag);
 	ShortSeq = new(DevVarShortArray);
 	ShortSeq.inout() << datum;
+	data_type = DEV_SHORT;
```

Comment:
Looks to me that the whole constructor can be delegated to the *DeviceAttribute(const cha*)*:

```c++
DeviceAttribute::DeviceAttribute(string &new_name, short datum):ext(new DeviceAttributeExt)
{
    this(new_name.c_str(), datum);
}
```

Is this available in C++? Otherwise a helper method may be introduced.